engine/execution/ingestion/engine.go

```diff
@@ -1078,7 +1078,7 @@ func (e *Engine) saveExecutionResults(
 			collectionID = flow.ZeroID
 		}
 
-		chunk := generateChunk(i, startState, endState, collectionID, blockID)
+		chunk := generateChunk(i, startState, endState, collectionID, blockID, result.Events[i].Hash())
```

Comment:
I don't think it is a good idea to do the operation here, this is not a negligible operation and directly impact TPS slowdown. We probably should do the hashing operation as part of committer and just pass the final root hash to generateChunk.

Reply:
That was planned as a next step, to keep this PR simple

Reply:
My only concern is if this gets merge into master before the next step and make it to an spork

Reply:
Sure, I can prepare PR on top of this to before merging this one

Reply:
https://github.com/onflow/flow-go/pull/868